client/src/components/productOverview/ProductOverviewContainer.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { connect } from 'react-redux';
+// import { bindActionCreators } from 'redux';
+import ImageGallery from './ImageGallery.jsx';
+import fetchProduct from '../../actions/fetchProduct';
+
+class ProductOverviewContainer extends React.Component {
+
+  componentDidMount() {
+    this.props.handleFetch();
+  }
```

Comment:
This is not blocking just sth I've been thinking about: for ratings & reviews, I would need to call `/reviews` in `componentDidMount` from the reviews component, so basically we have multiple requests sent from multiple `componentDidMount`, would that cause the UI to behave weirdly? like if one request get the response first, that section will render first but the other sections may still be loading. Would this be a concern?

Reply:
This could definitely be a concern and one that will might have an evolving answer as we go. Could definitely cause performance issues. I'll need to do some more research to see what might be optimal. React router might give us a better solution if we end up using it.